### PR TITLE
Remove unnecessary embed precompiled framework phase

### PR DESCRIPTION
**Pull request by mstfy**
Removes unnecessary embed precompiled frameworks build phase when there is no framework to embed.

### CODE REVIEW

File: Sources/TuistGenerator/Generator/LinkGenerator.swift [hunk 1]
```diff
@@ -208,6 +208,8 @@ final class LinkGenerator: LinkGenerating {
     ) throws {
         let embeddableFrameworks = graphTraverser.embeddableFrameworks(path: path, name: target.name).sorted()
 
+        guard !embeddableFrameworks.isEmpty else { return }
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/LinkGenerator.swift:211]: There are cases in which embeddableFrameworks is not empty, but precompiledEmbedPhase ends up being empty (e.g. all embeddableFrameworks are library, bundle, or sdk).
This logic should be plugged in the if branch where we set the script to `"echo \"Skipping, nothing to be embedded.\""`

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: Sources/TuistGenerator/Generator/LinkGenerator.swift (hunk 1 above)
Comment (kwridan) [reply Sources/TuistGenerator/Generator/LinkGenerator.swift:211]: Indeed the distinction is by the nature of the dependency - the results of `embeddableFrameworks` contain a mixture of different dependency types, here are some of the interesting ones:

- `.product()`: A native Xcode target which can be embedded in the "Embed Frameworks"
- `.xcfamework`: External pre-compiled xcframeworks, Xcode knows how to deal with this and can also be added to "Embed Frameworks"
- `.framework`: External pre-compiled `.framework` dependency which goes to the the special script phase _(to strip out the unnecessary slices - this is the methodology that is typically used with Carthage frameworks)_

As @danyf90 only a subset go to the pre-complied phase. 

May be worth extending the test case to cover this?

**Review by kwridan — COMMENTED** [review]
Thanks @mstfy !

**Review by fortmarek — APPROVED** [review]
Thanks for writing the test @mstfy, this looks good!

File: Sources/TuistGenerator/Generator/LinkGenerator.swift [hunk 2]
```diff
@@ -232,6 +224,7 @@ final class LinkGenerator: LinkGenerating {
                 // Do nothing
                 break
             case .bundle:
+                // Do nothing
                 break
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/LinkGenerator.swift:228]: Can we group all the do nothing in the same case at the bottom?
e.g.
```
case .library, .bundle, .sdk:
    // Do nothing
    break
```

File: Sources/TuistGenerator/Generator/LinkGenerator.swift [hunk 3]
```diff
@@ -273,7 +266,13 @@ final class LinkGenerator: LinkGenerating {
             precompiledEmbedPhase.inputPaths = script.inputPaths
                 .map { "$(SRCROOT)/\($0.pathString)" }
             precompiledEmbedPhase.outputPaths = script.outputPaths
+
+            pbxproj.add(object: precompiledEmbedPhase)
+            pbxTarget.buildPhases.append(precompiledEmbedPhase)
         }
+
+        pbxproj.add(object: embedPhase)
+        pbxTarget.buildPhases.append(embedPhase)
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/LinkGenerator.swift:275]: Shouldn't this be done only inside the if? If yes, I would make that `if` a `guard` instead

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: Sources/TuistGenerator/Generator/LinkGenerator.swift (hunk 3 above)
Comment (mstfy) [reply Sources/TuistGenerator/Generator/LinkGenerator.swift:275]: Actually there are 2 build phases in here. `embedPhase` and `precompiledEmbedPhase`. The one needs checking is `precompiledEmbedPhase` and it is added in if statement.

**Review by danyf90 — APPROVED** [review]

**Pull request merged**